submissions/OlStaniev/OOP exercise/index.js

```diff
@@ -0,0 +1,42 @@
+class Inhabitants {
+   constructor(name, gender, legs, saying) {
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying
+   }
+
+   prepareToPrint() {
+      return [this.name, this.gender, this.legs, this.saying].join(';')
+   }
+}
+
+class People extends Inhabitants {
+   constructor(name, gender, legs, hands, saying) {
+      super(name, gender, legs, saying)
+      this.hands = hands
+   }
+   prepareToPrint() {
+      return [this.name, this.gender, this.legs, this.hands, this.saying].join(';')
+   }
+}
+
+class PeopleWithFriends extends People {
+   constructor(name, gender, legs, hands, saying, friends) {
+      super(name, gender, legs, hands, saying)
+      this.friends = friends
+   }
+   prepareToPrint() {
+      return [this.name, this.gender, this.legs, this.hands, this.saying, this.friends].join(';')
+   }
+}
+
+const dog = new Inhabitants('Bob', 'male', 4, 'woof!')
+const cat = new Inhabitants('Kitty', 'female', 4, 'meow!')
+const woman = new People('Sara', 'female', 2, 2, 'Hello!')
+const catWoman = new People('Bella', 'female', 2, 2, `${cat.saying}`)
+const man = new PeopleWithFriends('Mario', 'male', 2, 2, 'Hi!', ['Bob', 'Tom', 'Eva'])
```

Comment:
By convention class name should be in singular form unless it holds multiple entitites indeed. `new People` that returns single human reads a bit weird as there is no option to create multiple humans in a single call of `new People`.

No need to have `People` and `PeopleWithFriends`. This can be resolved by having default `friends = []` in a class constructor signature and relevant logic in `prepareToPrint` method. Even if `friends` has no default value and assigned with none, it's value gonna be `undefined`, which is also quite handleable.

Imagine we want to specify neighbours, pets etc (and any such list may be empty or undefined). This should be handled by a single class.